autode/values.py

```diff
@@ -65,23 +68,21 @@ def _to(value: Union["Value", "ValueArray"], units: Union[Unit, str]):
             f"No viable unit conversion from {value.units} -> {units}"
         )
 
-    # Convert to the base unit, then to the new units
-    c = float(units.conversion / value.units.conversion)
-
-    if isinstance(value, Value):
-        return value.__class__(float(value) * c, units=units)
-
-    elif isinstance(value, ValueArray):
-        value[:] = np.array(value, copy=True) * c
-        value.units = units
-        return value
-
-    else:
+    if not (isinstance(value, Value) or isinstance(value, ValueArray)):
         raise ValueError(
             f"Cannot convert {value} to new units. Must be one of"
             f" Value of ValueArray"
         )
 
+    # Convert to the base unit, then to the new units
+    c = float(units.conversion / value.units.conversion)
+
+    new_value = value if inplace else value.copy()
+    new_value *= c
+    new_value.units = units
+
+    return None if inplace else new_value
```

Comment:
I just tested this and it only works for ValueArray, not value.
```ipython
In [4]: from autode.values import Energy

In [5]: x = Energy(0.5, 'Ha')

In [6]: x
Out[6]: Energy(0.5 Ha)

In [7]: new_x = x

In [8]: new_x *= 4

In [9]: new_x
Out[9]: Energy(2.0 Ha)

In [10]: x
Out[10]: Energy(0.5 Ha)
```
I think modifying a value (like Energy) always creates a copy because they are derived from float and floats are immutable in python so `__new__` is always called. I don't think an inplace edit is possible for `value`. But inplace is possible for valuearray as numpy array object mutable.

I also did some more tests and it turns out that multiplication also removes any attributes from the original `value` object as it also creates a new object by calling `self.__class__(...)`. So now I think the previous code was fine. For the valuearray, I think doing copy() and then multiplying when not inplace would be better. (Sorry for my confusion 😅)

Reply:
> I think modifying a value (like Energy) always creates a copy because they are derived from float and floats are immutable in python so __new__ is always called. I don't think an inplace edit is possible for value. But inplace is possible for valuearray as numpy array object mutable.

yep absolutely. That's what I was expecting and I think should happen – `Value`~`float`, `ValueArray`~`np.ndarray`

 
>  also did some more tests and it turns out that multiplication also removes any attributes from the original value object as it also creates a new object by calling self.__class__(...)

excellent spot – thanks. I'll try and fix this...